File: Caesar Cypher Encryptor/caesarCypherEncryptor.cpp
#include <string>
#include <vector>
#include <numeric>

using namespace std;

char helper(char letter, int key);

string caesarCypherEncryptor(string str, int key)
{
    vector<char> newLetters;
    int newKey = key % 26;
    for (int i = 0; i < str.length(); i++)
    {
        newLetters.push_back(helper(str[i], newKey));
    }
    return string(newLetters.begin(), newLetters.end());
}

char helper(char letter, int key)
{
    int newLetterCode = letter + key;
    return newLetterCode <= 122 ? newLetterCode : 96 + newLetterCode % 122;
}

File: Caesar Cypher Encryptor/caesarCypherEncryptor.py
def caesarCypherEncryptor(string, key):
    newLetters = []
    newKey = key % 26
    for letter in string:
        newLetters.append(helper(letter, newKey))
    return "".join(newLetters)


def helper(letter, key):
    newLetterCode = ord(letter) + key
    return chr(newLetterCode) if newLetterCode <= 122 else chr(96 + newLetterCode % 122)
